airbyte-lib/airbyte_lib/factories.py

```diff
@@ -3,14 +3,47 @@
 
 from typing import Any, Dict, Optional
 
-from airbyte_lib.cache import InMemoryCache
+import ulid
+from airbyte_lib.caches.duckdb import DuckDBCache, DuckDBCacheConfig
 from airbyte_lib.executor import PathExecutor, VenvExecutor
 from airbyte_lib.registry import get_connector_metadata
 from airbyte_lib.source import Source
+from airbyte_protocol.models import ConfiguredAirbyteCatalog
 
 
-def get_in_memory_cache():
-    return InMemoryCache()
+def get_default_cache(source_catalog: ConfiguredAirbyteCatalog) -> DuckDBCache:
+    """Get a local cache for storing data, using the default database path.
+
+    Cache files are stored in the `.cache` directory, relative to the current
+    working directory.
+    """
+    config = DuckDBCacheConfig(
+        db_path="./.cache/default_cache_db.duckdb",
+    )
+    return DuckDBCache(
+        config=config,
+        source_catalog=source_catalog,
+    )
+
+
+def new_local_cache(
+    cache_name: str | None = None,
```

Comment:
This behavior is intentional, specifically to avoid side effects of previous runs. When preferring accessibility and re-use, the user should give a name or use the similar factory function `get_default_cache()` which performs as you describe.

The proposed cleanup method (if/when needed) would be to simply delete the parent directory if remnants get out of hand. We use a cache directory which is a subdirectory to `cwd` in order to make this easy to explore and/or cleanup as needed.